sdk/nodejs/resource.ts

```diff
@@ -673,7 +677,7 @@ export interface ResourceOptions {
      *
      * If this is a [ComponentResourceOptions] do not provide both [provider] and [providers]
      */
-    provider?: ProviderResource;
+    provider?: ProviderResourceReference;
```

Comment:
Observable change that can lead to compile-time errors if user code *consumes* the value. Should not break programs that simply populate the value, which is more expected.